src/GameServer/MessageHandler/TargetedSkillHandlerPlugIn.cs

```diff
@@ -13,21 +13,41 @@ namespace MUnique.OpenMU.GameServer.MessageHandler;
 using MUnique.OpenMU.PlugIns;
 
 /// <summary>
-/// Handler for targeted skill packets.
+/// Reprersents the targeted skill packet handler.
+/// </summary>
+internal partial class TargetedSkillHandlerPlugIn
+{
+    private readonly Dictionary<int, ITargetedSkillAction> _handlers = [];
+    private readonly ITargetedSkillAction _defaultHandler = new TargetedSkillActionDefault();
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TargetedSkillHandlerPlugIn"/> class.
+    /// </summary>
+    public TargetedSkillHandlerPlugIn()
+    {
+        // Load all plugins in the current assembly
+        var pluginType = typeof(ITargetedSkillAction);
+        var pluginInstances = AppDomain.CurrentDomain.GetAssemblies()
+            .SelectMany(a => a.GetTypes())
+            .Where(t => pluginType.IsAssignableFrom(t) && !t.IsInterface && !t.IsAbstract)
+            .Select(t => (ITargetedSkillAction)Activator.CreateInstance(t)!)
+            .ToList();
+
+        foreach (var plugin in pluginInstances)
+        {
+            this._handlers[plugin.Key] = plugin;
+        }
+    }
```

Comment:
The plugin framework can do similar stuff already. You'd probably want to implement strategy plugins. To get the plugin manager, you can go over `player.GameContext.PlugInManager`. I think there is similar stuff for the area skills.

Reply:
Updated to use the PlugInManager as suggested.